### PR TITLE
feat: release v3 by merging next into main

### PR DESCRIPTION
**Pull request by muhlemmer**
This branch is an intermediate state where next and main are merged together to solve merge conflicts. After the conflicts were resolved the following changes were made:

1. Regenerate proto stubs to match the latest version from zitadel
2. Update all go modules, so we make sure no security advisories stay active
3. Update release info in readme.

BREAKING CHANGES

See past commits from the `next` branch. This header will trigger v3 release.

Note to merger:  This PR **must not** be squashed in order to keep proper commit history.

Closes https://github.com/zitadel/zitadel-go/issues/348

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by muhlemmer** [comment]
As `next` dropped support for go <1.21 we need to disable the test requirement for v1.20

File: example/auth/auth.go [hunk 1]
```diff
@@ -20,16 +21,19 @@ var (
...
 	//- api (e.g. acme-dtfhdg.zitadel.cloud:443)
 	//- scopes (including the ZITADEL project ID),
 	//- a JWT Profile source token (e.g. path to your key json), if not provided, the file will be read from the path set in env var ZITADEL_KEY_PATH
 	client, err := auth.NewClient(
+		ctx,
 		*issuer,
 		*api,
 		[]string{oidc.ScopeOpenID, zitadel.ScopeZitadelAPI()},
-		//zitadel.WithJWTProfileTokenSource(middleware.JWTProfileFromPath("key.json")),
+		zitadel.WithJWTProfileTokenSource(middleware.JWTProfileFromPath(ctx, "key.json")),
```
Comment (adlerhurst) [inline example/auth/auth.go:36]: would it make sense to add the path as flag with default value key.json?

**Review by adlerhurst — COMMENTED** [review]
not sure if the code for the NewClient functions would get more readable if we would use a helper function instead of copy pasting the code.

Instead of the protoc stuff in the build/zitadel folder would it make builds easier if we switch to buf?

File: example/auth/auth.go (hunk 1 above)
Comment (muhlemmer) [reply example/auth/auth.go:36]: It's just an example. Anyone who follows the example is free how to configure the path. Flag, yaml, viper...  Would the example benefit from such additional logic?

**Comment by muhlemmer** [comment]
@adlerhurst 

> not sure if the code for the NewClient functions would get more readable if we would use a helper function instead of copy pasting the code.
> 
> Instead of the protoc stuff in the build/zitadel folder would it make builds easier if we switch to buf?

I agree. But such a change should be formalized in a user-story and planned. The scope of this PR is merging the current state of next into main, which was neglected for quite some months now.

File: example/auth/auth.go (hunk 1 above)
Comment (adlerhurst) [reply example/auth/auth.go:36]: i don't think so.

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**